### ISSUE TITLE
Fault receivers are not printed (SeisSol GPU)

### ISSUE THREAD

**Issue by francescomosconi** [SeisSol/SeisSol#693]
**Describe the bug**
When running with `ClusteredLTS = 2` the code doesn't print the output for the faultreceivers file, except for the 0s time.

<img width="1170" alt="Schermata 2022-10-10 alle 15 25 58" src="https://user-images.githubusercontent.com/84501864/194877007-555543c7-5cb9-453a-a300-9ea5cbfc16b8.png">

**Expected behavior**
Expect to print the faultreceivers output for every time step

**To Reproduce**
1. Used version ` 202103_Sumatra-1090-g7ae55cfa  `   The branch is ravil/energy-output 
2. Build setting for GPU with CUDA. SeisSol_RelWithDebInfo_ssm_70_cuda_6_elastic. With MPI compilers.
3. On which machine does your problem occur? If on a cluster: Which modules are loaded?
   Marconi100.   

<img width="750" alt="Schermata 2022-10-10 alle 15 40 10" src="https://user-images.githubusercontent.com/84501864/194879628-9fef0976-758c-4546-92b3-926f1d9c8548.png">

4. Provide parameter/material files.
[parameters.txt](https://github.com/SeisSol/SeisSol/files/9746405/parameters.txt)

Many thanks in advance.

**Comment by AliceGabriel** [comment]
Hi just following up on this, @francescomosconi did you check if your receivers are placed correctly? How to do this is explained here https://github.com/SeisSol/Meshing/tree/master/place_receivers

**Comment by francescomosconi** [comment]
Hi @AliceGabriel, thanks for your suggestion. 
I think that with the positioning of the fault receivers every things work correctly.

In the seissol.out file there is the voice `all point receivers found along the fault`. 

More confirmation comes from the fact that fault receivers are only printed at time 0s with the correctly parameters, but it goes no further than that.

These are correctly printed, for all the calculation time steps, just when it is being used `ClusteredLTS=1`.

**Comment by Thomas-Ulrich** [comment]
Hi,

I can confirm that the problem, both on CPU (NG) and GPUs (heisenbug).

**Comment by Thomas-Ulrich** [comment]
The problem is here:
https://github.com/SeisSol/SeisSol/blob/master/src/Solver/time_stepping/TimeCluster.cpp#L698
If there is no DR faces in first cluster, this does not work.

**Comment by ravil-mobile** [comment]
> The problem is here: https://github.com/SeisSol/SeisSol/blob/master/src/Solver/time_stepping/TimeCluster.cpp#L698 If there is no DR faces in first cluster, this does not work.

Thanks @Thomas-Ulrich, now I can see where the problem stems from. Do you have any idea about a fix?

**Comment by Thomas-Ulrich** [comment]
I hoped you had an idea on how to fix that :-)
I would introduce and initialize a variable `m_isFirstDynamicRuptureCluster` and use that instead of 
`if (m_clusterId == 0`. Would it be doable?

**Comment by AliceGabriel** [comment]
I think @krenzland may be the best to answer this?

**Comment by ravil-mobile** [comment]
@Thomas-Ulrich, I guess we need to find the first non-zero DR cluster and used it instead of the hard-coded `0`. I guess we also need to adjust `dynamicRuptureScheduler::mayComputeFaultOutput`. I can have a look at it

**Comment by krenzland** [comment]
Yeah, if you cannot do it / do not want to do it, ping me next week :) I think it's easy to fix.

**Comment by francescomosconi** [comment]
Hi, any update on that?

**Comment by krenzland** [comment]
Please try #753. I haven't tested it this much but it should (!) fix the issue. If it doesn't, @ravil-mobile probably needs to have a more in-depth look.
